crates/ty_server/src/session/index.rs

```diff
@@ -83,26 +59,88 @@ impl Index {
         new_version: DocumentVersion,
         encoding: PositionEncoding,
     ) -> crate::Result<()> {
-        // update notebook cell index
-        if let Some(lsp_types::NotebookDocumentCellChangeStructure {
-            did_open: Some(did_open),
-            ..
-        }) = cells.as_ref().and_then(|cells| cells.structure.as_ref())
-        {
-            for opened_cell in did_open {
-                let cell_path = SystemVirtualPath::new(opened_cell.uri.as_str());
-                self.notebook_cells
-                    .insert(cell_path.to_string(), notebook_key.to_file_path());
-            }
-            // deleted notebook cells are closed via textDocument/didClose - we don't close them here.
-        }
-
         let document = self.document_mut(notebook_key)?;
         let Some(notebook) = document.as_notebook_mut() else {
             anyhow::bail!("Notebook document path does not point to a notebook document");
         };
 
-        notebook.update(cells, metadata, new_version, encoding)?;
+        let (structure, data, text_content) = cells
+            .map(|cells| {
+                let lsp_types::NotebookDocumentCellChange {
+                    structure,
+                    data,
+                    text_content,
+                } = cells;
+                (structure, data, text_content)
+            })
+            .unwrap_or_default();
```

Comment:
Is there a specific reason for using `unwrap_or_default` here instead of `if let ...` ?